lib/dec2oct.js

```diff
@@ -1,7 +1,21 @@
 // Write a function that returns octal state of decimal number
 
 function dec2oct(a) {
-  // your code here
+  var number = a;
+  var octal = "";
+  var remainder;
+
+  if (number === 0) {
+    octal = 0;
+  } else {
+    while(number > 0) {
+      remainder = number % 8;
+      octal = remainder + octal;
+      number = Math.floor(number / 8);
+    }
+  }
+
+  return octal;
```

Comment:
Great work on this. You may wanna have a look at the `toString` method. It has the ability to convert numbers to different bases. `parseInt` can parse a number from a string in many bases too.